rpc/src/lib.rs

```diff
@@ -231,8 +231,27 @@ impl<B, C, SC, P, CT, BE> EthApiT for EthApi<B, C, SC, P, CT, BE> where
 		   .map_err(|_| internal_err("fetch runtime account basic failed"))?.nonce.into())
 	}
 
-	fn block_transaction_count_by_hash(&self, _: H256) -> BoxFuture<Option<U256>> {
-		unimplemented!("block_transaction_count_by_hash");
+	fn block_transaction_count_by_hash(&self, hash: H256) -> Result<Option<U256>> {
```

Comment:
I think this `hash` is Ethereum block hash, right? Then feeding that into `BlockId::Hash(hash)` will not work because that's supposed to be a Substrate block hash.

I think we should directly use the runtime api to fetch the Ethereum block and get transaction count from there.

Reply:
Oh yes, totally right